frontend/public/components/user.tsx

```diff
@@ -75,22 +84,38 @@ const oAuthResourcePath = resourcePathFromModel(OAuthModel, 'cluster');
 
 const NoDataEmptyMsg = () => {
   const { t } = useTranslation();
+  const canEditIdentityProviders = useCanEditIdentityProviders();
+  const [oauth, oauthLoaded] = useOAuthData(canEditIdentityProviders);
   return (
     <MsgBox
       title={t('public~No Users found')}
       detail={
-        <>
+        canEditIdentityProviders && oauthLoaded ? (
+          oauth?.spec?.identityProviders?.length > 0 ? (
+            <p>
+              <UsersHelpText />
+            </p>
+          ) : (
+            <>
+              <p>
+                {t(
+                  'public~Add identity providers (IDPs) to the OAuth configuration to allow others to log in.',
+                )}
+              </p>
+              <p>
+                <Link to={oAuthResourcePath}>
+                  <Button variant="primary">{t('public~Add IDP')}</Button>
+                </Link>
+              </p>
+            </>
```

Comment:
Since we've made `UsersHelpText` a separate component, I'd make this its own component as well.